### PR TITLE
Fix the agent installation command for macOS

### LINKED ISSUE #4953: MacOS agent installation command fails due to `launchctl setenv`
| Wazuh | Elastic | Rev  | Security |
| ----- | ------- | ---- | -------- |
| 4.4-alpha1   | 7.17     | 4400 | Basic, ODFE, Xpack |

| Browser |
| ------- |
| Chrome, Firefox, Safari, etc|

**Description**

During E2E testing in [this issue](https://github.com/wazuh/wazuh/issues/15555#issuecomment-1333557857), a problem appeared when trying to install an agent using the command suggested in the WUI:
image

```
$ curl -so wazuh-agent-4.4.0.pkg https://packages-dev.wazuh.com/pre-release/macos/wazuh-agent-4.4.0-1.pkg && sudo launchctl setenv WAZUH_MANAGER '192.168.1.140' WAZUH_AGENT_GROUP 'default' WAZUH_AGENT_NAME='macos' && sudo installer -pkg ./wazuh-agent-4.4.0.pkg -target /
Usage: launchctl setenv <<key> <value>, ...>
```

That bug was already reported in the past in issues like this:
- https://github.com/wazuh/wazuh-kibana-app/issues/3211

**Preconditions**
None

**Steps to reproduce**
1. Navigate to `Add agent`
2. Click on macOS
3. Click on Mojave
4. Click on Intel 
5. Set IP addres 
6. Select group 
7. Copy suggested command
8. Paste it on macOS agent.

**Expected Result**
1. The environment variables are set correctly and the agent is installed.

**Actual Result**
1. Environment variables cannot be set and the package is not installed.

### PR DESCRIPTION
**Pull request by chantal-kelm**
### Description

This PR fixes the agent installation command for macOS

This pull request is going to be merged into the main branch used for this epic [#4931](https://github.com/wazuh/wazuh-kibana-app/pull/4933)

### Manual Backports

[4.4-2.3-wzd](https://github.com/wazuh/wazuh-kibana-app/pull/4982)
[4.4-7.16 ](https://github.com/wazuh/wazuh-kibana-app/pull/4983)

### Issues Resolved

#4953 

### Evidence

image

### Steps to test

* Click on agents
* Click deploy new agent
* Follow the steps to deploy a new macOS agent
* In the step 'Install and enroll agent' the variables WAZUH_MANAGER, WAZUH_AGENT_GROUP and WAZUH_AGENT_NAME *should not contain a '=' sign.

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by davidjiglesias — COMMENTED** [review]
There is spanish in the changelog, please fix.

**Review by Mayons95 — DISMISSED** [review]
TR: :heavy_check_mark: 

LGTM

Screenshot from 2022-12-15 11-05-33

File: public/controllers/agent/components/register-agent.js [hunk 1]
```diff
@@ -367,6 +367,14 @@ export const RegisterAgent = withErrorBoundary(
 
     agentNameVariable() {
       let agentName = `WAZUH_AGENT_NAME='${this.state.agentName}' `;
+      if (
+        this.state.selectedOS === 'macos' &&
+        this.state.selectedArchitecture &&
+        this.state.agentName !== ''
+      ) {
+        return agentName.replace(/=/g, ' ');
```
Comment (Desvelao) [inline public/controllers/agent/components/register-agent.js:375]: **thought**: I dislike this solution, replacing the character. We should build the variable declaration for the different use cases and not build for one of them and then change it for a particular case. 

I guess we could improve it with a possible refactoring of the entire agent deployment guide.

**Review by Desvelao — DISMISSED** [review]
**review**

:heavy_check_mark: code
:heavy_check_mark: code

Screenshot
image

**Pull request merged**